### PR TITLE
fix: improve profiles page load by ~8s with 200 more routes

### LINKED ISSUE #139: Slowness on client
Expanding routes

Filtering routes

Is mezzo client being instantiated each time? Logs indicate so

### PR DESCRIPTION
**Pull request by sdg9**
Closes #139

One home page (routes) every time you expanded or collapsed a single item ALL routes would re-render.  This was visually evident with 200ish routes.  I resolved by memoizing the setState callback and instead of passing the selected route string to ALL components (changed every selection), I pulled the conditional check up to the routes page so that it only passed a boolean to each component (so that ONLY the component(s) which changed will rerender).

On profile page instead of making api call to select all routes I just make api call to get active variants, then I perform check against that.  This helped reduce time it took to process especially when there were hundreds of routes by several seconds.

### CODE REVIEW

File: apps/admin-web/src/app/app.tsx [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
 import ProfilesScreen from './components/Profiles/ProfilesScreen';
+import { ClientContext } from './context';
+import mezzoClient from '@caribou-crew/mezzo-core-client';
 
 // If prod, routes are /mezzo and /mezzo/record instead of / and /record
 export const App = () => {
+  const client = mezzoClient();
+  // const MyContext = React.createContext(client);
+
+  console.log('=============RERENDER APP ROOT');
```
Comment (NSwanson1) [inline apps/admin-web/src/app/app.tsx:18]: Is this being kept intentionally?

File: apps/admin-web/src/app/components/HomeScreen.tsx [hunk 2]
```diff
@@ -164,29 +167,49 @@ export default function Home(props: Props) {
...
-          )}
-        >
-          {renderRouteList()}
-        </Flipper>
+
+      {isLoading ? (
+        <PacmanLoader
+          // color={'#ea82e5'}
+          // color={'#46bfee'}
+          // color={'#d03e19'}
```
Comment (NSwanson1) [inline apps/admin-web/src/app/components/HomeScreen.tsx:175]: can you remove these commented colors.

File: apps/admin-web/src/app/components/Profiles/ProfileDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+          <Typography variant="subtitle1">{profile.name}</Typography>
+        </Box>
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'row',
+            alignItems: 'center',
+          }}
+        >
+          {/* {type === 'local' && _renderDeleteLocalProfileButton(profile)} */}
```
Comment (NSwanson1) [inline apps/admin-web/src/app/components/Profiles/ProfileDetails.tsx:75]: Why is this commented?

File: apps/admin-web/src/app/components/Profiles/ProfilesScreen.tsx [hunk 4]
```diff
@@ -1,50 +1,32 @@
...
+import SaveAsRemote from './SaveAsRemote';
 
 SyntaxHighlighter.registerLanguage('javascript', js);
 
 type Props = Record<string, never>;
 
 export default function ProfilesScreen(props: Props) {
-  const client = mezzoClient();
-  const { routes } = useFetchRoutes();
+  console.log('Profile Screen');
```
Comment (NSwanson1) [inline apps/admin-web/src/app/components/Profiles/ProfilesScreen.tsx:22]: this can be removed

File: apps/admin-web/src/app/app.tsx (hunk 1 above)
Comment (sdg9) [reply apps/admin-web/src/app/app.tsx]: fixed

File: apps/admin-web/src/app/components/HomeScreen.tsx (hunk 2 above)
Comment (sdg9) [reply apps/admin-web/src/app/components/HomeScreen.tsx]: fixed

File: apps/admin-web/src/app/components/Profiles/ProfileDetails.tsx (hunk 3 above)
Comment (sdg9) [reply apps/admin-web/src/app/components/Profiles/ProfileDetails.tsx]: removed

File: apps/admin-web/src/app/components/Profiles/ProfilesScreen.tsx (hunk 4 above)
Comment (sdg9) [reply apps/admin-web/src/app/components/Profiles/ProfilesScreen.tsx]: removed

**Review by NSwanson1 — APPROVED** [review]

**Pull request merged**